src/types.jl

```diff
@@ -1,143 +1,219 @@
+            acceleration = CPU1(),
+            embedding_dims = Dict{Symbol, Real}(),
+        ) where {B, O, L}
+
+            model = $Model{B, O, L}(
+                builder,
+                optimiser,
+                loss,
+                epochs,
+                batch_size,
+                lambda,
+                alpha,
+                rng,
+                optimiser_changes_trigger_retraining,
+                acceleration,
+                embedding_dims,
+            )
+
+            message = clean!(model)
+            isempty(message) || @warn message
+
+            return model
+        end
+
+    end |> eval
 
 end
 
 const Regressor =
-  Union{NeuralNetworkRegressor,MultitargetNeuralNetworkRegressor}
+    Union{NeuralNetworkRegressor, MultitargetNeuralNetworkRegressor}
+
+
+# Separately define the ImageClassifier
+mutable struct ImageClassifier{B, F, O, L} <: MLJFluxProbabilistic
+    builder::B
+    finaliser::F
+    optimiser::O
+    loss::L
+    epochs::Int
+    batch_size::Int
+    lambda::Float64
+    alpha::Float64
+    rng::Union{AbstractRNG, Int64}
+    optimiser_changes_trigger_retraining::Bool
+    acceleration::AbstractResource
+end
+
+function ImageClassifier(
+    ; builder::B = image_builder(VGGHack),
+    finaliser::F = Flux.softmax,
+    optimiser::O = Optimisers.Adam(),
+    loss::L = Flux.crossentropy,
+    epochs = 10,
+    batch_size = 1,
+    lambda = 0,
+    alpha = 0,
+    rng = Random.default_rng(),
+    optimiser_changes_trigger_retraining = false,
+    acceleration = CPU1(),
+) where {B, F, O, L}
+    model = ImageClassifier{B, F, O, L}(
+        builder,
+        finaliser,
+        optimiser,
+        loss,
+        epochs,
+        batch_size,
+        lambda,
+        alpha,
+        rng,
+        optimiser_changes_trigger_retraining,
+        acceleration,
+    )
+
+    message = clean!(model)
+    isempty(message) || @warn message
+
+    return model
+end
+
 
 MMI.metadata_pkg.(
-  (
-      NeuralNetworkRegressor,
-      MultitargetNeuralNetworkRegressor,
-      NeuralNetworkClassifier,
-      ImageClassifier,
-      NeuralNetworkBinaryClassifier,
-  ),
-    name="MLJFlux",
-    uuid="094fc8d1-fd35-5302-93ea-dabda2abf845",
-    url="https://github.com/alan-turing-institute/MLJFlux.jl",
-    julia=true,
-    license="MIT",
+    (
+        NeuralNetworkRegressor,
+        MultitargetNeuralNetworkRegressor,
+        NeuralNetworkClassifier,
+        ImageClassifier,
+        NeuralNetworkBinaryClassifier,
+    ),
+    name = "MLJFlux",
+    uuid = "094fc8d1-fd35-5302-93ea-dabda2abf845",
+    url = "https://github.com/alan-turing-institute/MLJFlux.jl",
+    julia = true,
+    license = "MIT",
 )
 
+const MODELSUPPORTDOC = """
+This model supports categorical columns in the input table. If present, such columns are embedded into
+dense vectors by the use of an additional `EntityEmbedder` layer after the input 
+as described in Entity Embeddings of Categorical Variables by Cheng Guo, Felix Berkhahn.
```

Comment:
Please check to replace "columns" in document strings with "features". This is for consistency with other MLJ documentation and also future proofs for the case (in the distant future) where matrix input will likely have "feaures=rows" (because that is what MLUtils.jl does: in tables features are columns, but in arrays the observation index is always the last one).

```suggestion
In addition to features with `Continuous` scientific element type, this model supports 
categorical features in the input table. If present, such features are embedded into
dense vectors by the use of an additional `EntityEmbedder` layer after the input 
as described in Entity Embeddings of Categorical Variables by Cheng Guo, Felix Berkhahn.
```

Reply:
@ablaom I must highlight that it crossed my mind and I was going to do it but I found that the original documentation used columns. See https://fluxml.ai/MLJFlux.jl/dev/interface/Regression/

Reply:
Done in b0d8e44ad734d23643c271cb373d1803a0b4b126